### DISCUSSION TITLE
returns incorrect password during login even when the server side log is returning '- password is correct'

### DISCUSSION THREAD

**Discussion by anusha928** [serverpod/serverpod#2509] [Q&A] [answered]
-returns incorrect password during login even when the server side log is returning '- password is correct'
-after entering validation code it returns invalid code but is adding user in userinfo table

Also the user is already in the user info table and the server side log is as :
Webserver listening on port 8082
METHOD CALL: serverpod_auth.status.getUserInfo duration: 18ms numQueries: 0 authenticatedUser: 1
DEBUG: authenticate chatterpod@gmail.com / XXXXXXXX
DEBUG:  - found entry 
DEBUG:  - password is correct, userId: 1)
METHOD CALL: serverpod_auth.email.authenticate duration: 266ms numQueries: 0 authenticatedUser: null

**Expected behavior**
Expected to login successfully with valid password

**Serverpod versions**
SERVERPOD version: 1.2.7, dart: 3.3.3 (stable) (Tue Mar 26 14:21:33 2024 +0000) 
mode: development, role: monolith, logging: normal, serverId: default
Insights listening on port 8081
Server default listening on port 8080

**Platform information**
on "linux_x64", time: 2024-05-23 09:12:18.826041Z

**Comment by Katekko** [comment]
Good morning,

Could you provide the repository for testing? If it's private, it would be helpful if you could create a small example instead.

**Comment by SandPod** [comment, marked as answer]
Hello!

Do you have access to the user email and password? If so, instead of using the authentication module, you could debug the error by calling the authentication endpoint directliy from your app. The endpoint will give a response that should contain the informtion about why the authentication was rejected.

You can call it using `var authResponse = await client.modules.auth.email.authenticate(email, password);`
